code/game/machinery/washing_machine.dm

```diff
@@ -1,54 +1,115 @@
 
-/obj/machinery/washing_machine/verb/start()
-	set name = "Start Washing"
-	set category = "Object"
-	set src in oview(1)
+	// How much damage a single cycle should do, 20 is just enough damage to kill Ian
+	var/washing_damage = 20
+	// How long a single cycle should take
+	var/washing_time = 15 SECONDS
+	// How often to damage the occupant
+	var/damage_time = 1 SECONDS
+	// Sounds made when occupant is damaged
+	var/hit_sound = list('sound/weapons/genhit1.ogg','sound/weapons/genhit2.ogg','sound/weapons/genhit3.ogg')
+	// How long it'll take to resist out
+	var/resist_time = 5 SECONDS
 
-	if(!istype(usr, /mob/living)) //ew ew ew usr, but it's the only way to check.
-		return
+/obj/machinery/washing_machine/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/machinery/sleeper/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/washing_machine/AltClick(mob/living/user)
+	Wash(user)
 
-	if( state != 4 )
-		to_chat(usr, "The washing machine cannot run in this state.")
+/obj/machinery/washing_machine/proc/Wash(mob/living/user)
+	if(!istype(user))
+		return
+	if(stat & NOPOWER)
+		to_chat(usr, "<span class='notice'>[src] isn't powered.</span>")
 		return
+	if(running)
+		to_chat(usr, "<span class='notice'>[src] is already running.</span>")
+		return
+	if(open && bloody)
+		if(!full)
+			to_chat(user, "<span class='notice'>You start [src] to clean it out.</span>")
+		else
+			to_chat(user, "<span class='notice'>You start [src], covering the contents inside with blood.</span>") // THIS DOESNT WORK YET
+	else
+		if(!full)
+			to_chat(user, "<span class='notice'>[src] is empty, there is nothing to clean.</span>")
+			return
+
+	playsound(src, 'sound/machines/click.ogg', 25)
+	running = TRUE
+	open = FALSE
 
-	if( locate(/mob,contents) )
-		state = 8
+	var/mob/living/M = locate(/mob/living, contents)
+	if(M)
+		if(user == M)
+			to_chat(M, "<span class='warning'>You activate [src] and quickly close the door on yourself!</span>")
+		bloody = TRUE
+		update_icon()
+		add_attack_logs(user, M, "activated a washing machine with them inside")
+		to_chat(M, "<span class='userdanger'>Your body into the walls repeatedly as [src] spins!</span>")
+		for(var/I in 1 to (washing_time / damage_time))
+			if(prob(33))
+				M.emote("scream")
+			M.adjustBruteLoss(washing_damage / (washing_time / damage_time))
+			playsound(src, pick(hit_sound), 10)
+			sleep(damage_time)
+			I++
+		M.AdjustConfused(3 SECONDS)
 	else
-		state = 5
-	update_icon()
-	sleep(200)
-	for(var/atom/A in contents)
-		A.clean_blood()
+		update_icon()
+		for(var/I in 1 to (washing_time / damage_time))
+			playsound(src, 'sound/weapons/jug_filled_impact.ogg', 10)
+			sleep(damage_time)
+
+	if(!bloody) // if its filled with blood, its not gonna clean blood
+		for(var/atom/A in contents)
+			A.clean_blood()
 
-	//Tanning!
+	//Tanning! // why?
 	for(var/obj/item/stack/sheet/hairlesshide/HH in contents)
 		var/obj/item/stack/sheet/wetleather/WL = new(src)
 		WL.amount = HH.amount
 		qdel(HH)
```

Comment:
noting your comment, you may just want to completely remove this feature, it's kinda bloat and takes away from actual tanners :/